playlist.c

```diff
@@ -228,14 +245,15 @@ static bool playlist_path_equal(const char *real_path,
 
 /**
  * playlist_core_path_equal:
- * @real_core_path      : 'Real' search path, generated by path_resolve_realpath()
- * @entry_core_path     : Existing playlist entry 'core path' value
+ * @real_core_path               : 'Real' search path, generated by path_resolve_realpath()
+ * @entry_core_path              : Existing playlist entry 'core path' value
+ * @try_core_base_filename_match : If true, try matching core base file names
  *
  * Returns 'true' if real_core_path matches entry_core_path
  * (Taking into account relative paths, case insensitive
  * filesystems)
  **/
-static bool playlist_core_path_equal(const char *real_core_path, const char *entry_core_path)
+static bool playlist_core_path_equal(const char *real_core_path, const char *entry_core_path, bool try_core_base_filename_match)
```

Comment:
Let's pass a `const playlist_config_t *config` here (instead of `try_core_base_filename_match`), like we do with `playlist_path_equal()`

This means you need to add a new `bool relative_paths` member to `playlist_config_t` - which will be useful elsewhere...

Reply:
Ok done.
The new `relative_paths` will be automatically set by the `playlist_config_set_base_content_directory()` function